tests/MongoDB.Bson.Tests/Serialization/BsonDocumentBackedClassTests.cs

```diff
@@ -0,0 +1,30 @@
+﻿/* Copyright 2018-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System.Reflection;
+using MongoDB.Bson.Serialization;
+using MongoDB.Bson.TestHelpers;
+
+namespace MongoDB.Bson.Tests.Serialization
+{
+    public class BsonDocumentBackedClassTests
+    {
+    }
+
+    public static class BsonDocumentBackedClassReflector
+    {
+        public static IBsonDocumentSerializer _serializer(this BsonDocumentBackedClass obj) => (IBsonDocumentSerializer)Reflector.GetFieldValue(obj, nameof(_serializer), BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.FlattenHierarchy);
```

Comment:
I'm curious about the naming scheme: any reason `_serializer` is used for the name a public extension method? I've only seen _ as a leading character for private fields.

Reply:
Comments:

1. This is not public API, this is testing code
2. We name the extension method the same as the private field we are accessing via reflection
3. If C# had extension properties we would use that, but C# only has extension methods

Reply:
Acknowledged!